### DISCUSSION TITLE
I want to know if pyright can identify the following type error

### DISCUSSION THREAD

**Discussion by wydxyreh** [microsoft/pyright#9453] [Q&A] [answered]
I wrote a simple Python code:
```
def add(a: int, b: int) -> int:
    return a + b
c: int = add("a","2")
```
After trying to configure the pyright parameter ***"typeCheckingMode": "strict"***, I found that pyright cannot detect this error. I also tried many other configuration parameters, but none of them can identify this error.
I really hope to know if this is a problem with pyright itself, or if it is a problem with my configuration. If it is the latter, could you please advise on how to correctly configure the pyright parameters?

**Comment by erictraut** [comment, marked as answer]
Yes, pyright will identify this error even in basic type checking mode. 

Code sample in [pyright playground](https://pyright-play.net/?code=CYUwZgBAhswBRQFwQJYDsAuAaCAjZ6GAlBALQB8qmiAUBPRAE4gYCuja0EA1HjQMYFMEALzRYcAERRJWSQCZJRGkA)

There must be a problem with your configuration. 

Are you using the command-line version of pyright? The language server version?

**Reply by wydxyreh** [reply]
Thank you for your response. Your thoughts have inspired me. My Python environment is 3.11.4, and I checked the version of pyright that I installed via pip. I found that it was 1.1.388, while the latest version has reached 1.1.389. 
```
WARNING: there is a new pyright version available (v1.1.388 -> v1.1.389).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

pyright 1.1.388
```
After I updated pyright to the latest version using the command `pip install pyright==1.1.389`, it correctly detected the error, and the error type was `reportArgumentType`. My issue has been resolved.

However, there is one thing that I'm very interested in. I previously tested the 1.1.388 version of pyright many times and tried various parameter configurations, but I was unable to detect this type error. But after updating to 1.1.389, it can correctly detect the error. So, I'm curious whether this type detection is a new addition in 1.1.389, or if the 1.1.388 version was supposed to detect it but encountered some bugs. Currently, on Windows with Python 3.11.4, I can consistently reproduce this issue by manually specifying the pyright version as 1.1.388.

I look forward to your reply.

**Reply by erictraut** [reply]
I suspect that the version of pyright that you previously installed was misconfigured or corrupted. All versions of pyright (from 1.0.0 on) could have detected this type error. If you don't believe me, try the online pyright playground, which allows you to specify the version of pyright.

**Reply by wydxyreh** [reply]
I just tested it in the online environment, and it seems that there were some strange issues when I configured the local environment. Thank you for your answer.
